src/ui/ui_idle_menu_nbgl.c

```diff
@@ -0,0 +1,110 @@
+/*******************************************************************************
+ *   Tron Ledger Wallet
+ *   (c) 2022 Ledger
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+#ifdef HAVE_NBGL
+
+#include "os.h"
+#include "os_io_seproxyhal.h"
+#include "ui_idle_menu.h"
+#include "glyphs.h"
+#include "ux.h"
+#include "nbgl_page.h"
+#include "nbgl_use_case.h"
+#include "settings.h"
+
+enum {
+    SWITCH_ALLOW_TX_DATA_TOKEN = FIRST_USER_TOKEN,
+    SWITCH_ALLOW_CSTM_CONTRACTS_TOKEN,
+    SWITCH_ALLOW_HASH_TX_TOKEN
+};
+
+#define NB_INFO_FIELDS 2
+static const char* const infoTypes[] = {"Version", "Tron App"};
+static const char* const infoContents[] = {APPVERSION, "(c) 2022 Ledger"};
+
+#define NB_SETTINGS_SWITCHES 3
+#define SWITCH_IDX(token) (token - SWITCH_ALLOW_TX_DATA_TOKEN)
+static uint8_t settings[NB_SETTINGS_SWITCHES] = {S_DATA_ALLOWED,S_CUSTOM_CONTRACT,S_SIGN_BY_HASH};
```

Comment:
```suggestion
static uint8_t settings[NB_SETTINGS_SWITCHES] = {S_DATA_ALLOWED, S_CUSTOM_CONTRACT, S_SIGN_BY_HASH};
```

Reply:
@xchapron-ledger there is a dedicated format PR : https://github.com/LedgerHQ/app-tron/pull/42/